### ISSUE TITLE
`chunks_exact_to_as_chunks` suggests iter() instead of iter_mut() for chunks_exact_mut and does not compile

### ISSUE THREAD

**Issue by awused** [rust-lang/rust-clippy#17593]
### Summary

`chunks_exact_to_as_chunks` suggests replacing `chunks_exact_mut(N)` with `as_chunks_mut::<N>().0.iter()` when it is (almost?) always wrong and does not compile, and should use `iter_mut()`.

The suggestion is also significantly longer and less obvious than the initial code.

### Reproducer

Code:

```rust
fn main() {
    let mut a = vec![1, 2, 3, 4, 5, 6];
    a.chunks_exact_mut(2).for_each(|c| c[0] = 2);
}
```

Current output:

```rust
a.as_chunks_mut::<2>().0.iter().for_each(|c| c[0] = 2);
```

Desired output:

```rust
a.as_chunks_mut::<2>().0.iter_mut().for_each(|c| c[0] = 2);
```

### Version

```text
rustc 1.98.0 (88d9e12ae 2026-08-18)
binary: rustc
commit-hash: 88d9e12ae178fab0fb5cc050a94da85685d449ea
commit-date: 2026-08-18
host: x86_64-unknown-linux-gnu
release: 1.98.0
LLVM version: 22.1.8
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by cuishuang** [comment]
@rustbot claim

I checked the current `master` implementation and found that mutable calls already use `iter_mut()`, as introduced by #17316. The Rust 1.98 snapshot from the report still contains the older implementation.

I'll prepare a focused PR adding exact UI/rustfix regression coverage for mutation through `for_each` and `map`/`collect`, without changing the broader const-generic behavior of this lint.
